### ISSUE TITLE
Performance benchmarks

### ISSUE THREAD

**Issue by s-dschro** [nodejs/undici#560]
Hey guys,

first of all, great work here!
I am interested in how this client does perform vs `axios` or `got`? Do you have any benchmarks?

**Comment by ronag** [comment]
`axios` and `got` are based on the core http client with some extra overhead so I would expect them to be slower than he `http - keepalive` entry below.

```js
http - keepalive x 12,028 ops/sec ±2.60% (265 runs sampled)
undici - pipeline x 31,321 ops/sec ±0.77% (276 runs sampled)
undici - request x 36,612 ops/sec ±0.71% (277 runs sampled)
undici - stream x 41,291 ops/sec ±0.90% (268 runs sampled)
undici - dispatch x 47,319 ops/sec ±1.17% (263 runs sampled)
```

**Comment by s-dschro** [comment]
This was also my assumption - but I guess more people would switch to this lib if there would be some form of comparison in the readme.

**Comment by ronag** [comment]
@s-dschro feel free to open a PR. But I don't think they are directly comparable. got and axios are more high level libraries. I would rather encourage got and axios to switch to undici under the good. @szmarczak

**Comment by Nytelife26** [comment]
@s-dschro When we become one of the first HTTP libraries to switch to undici under the hood for `@helperdiscord/centra` we'll let you know how it works out. It's already one of the fastest mainstream libraries for it on NPM, but hopefully switching to undici will push performance even further.

**Comment by Nytelife26** [comment]
@s-dschro Petitio is now powered by undici, meaning I can give you the results you're looking for (a direct benchmark comparison).

[Suite source](https://github.com/helperdiscord/http-benchmarks)

```
[CYCLE] got - promise x 1,856 ops/sec ±1.44% (573 runs sampled)
[CYCLE] request - callback x 4,484 ops/sec ±1.55% (569 runs sampled)
[CYCLE] node-fetch - promise x 4,225 ops/sec ±1.61% (570 runs sampled)
[CYCLE] centra - promise x 6,489 ops/sec ±1.04% (572 runs sampled)
[CYCLE] petitio - promise (undici) x 12,877 ops/sec ±1.00% (574 runs sampled)
[CYCLE] petitio - promise (http) x 7,205 ops/sec ±0.98% (571 runs sampled)
FASTEST: petitio - promise (undici)
SLOWEST: got - promise
```

TL;DR: Undici is fast. Capitalized. It's FAST!

**Comment by s-dschro** [comment]
Wow, that's pretty impressive.

**Comment by Nytelife26** [comment]
> Wow, that's pretty impressive.

Indeed it is. I hope many more libraries to come will begin to use undici - although, I'm not sure how long that will be since the disadvantage is needing to add a dependency which makes it look like the bundle size is *significantly* larger than the `http` versions.

Perhaps some day in the future undici will replace core `http`, but that's a long shot since this is all very new. Thank you for showing interest though, and if you have further benchmark requests or questions I'm happy to answer them.

As for including this in the README, there are already benchmarks present there, and as @ronag mentioned it would be unfair to compare undici directly to higher-level libraries. That isn't to say we couldn't link to other benchmarks or the like which showcase the difference between libraries using native `http` and undici, though (thoughts, team?)
